src/runtime/internal/utils.ts

```diff
@@ -55,33 +55,41 @@ export function isJsonRequest(event: H3Event) {
   );
 }
 
-export function normalizeError(error: any) {
+export function normalizeError(error: any, isDev?: boolean) {
   // temp fix for https://github.com/unjs/nitro/issues/759
   // TODO: investigate vercel-edge not using unenv pollyfill
   const cwd = typeof process.cwd === "function" ? process.cwd() : "/";
-  const stack = ((error.stack as string) || "")
-    .split("\n")
-    .splice(1)
-    .filter((line) => line.includes("at "))
-    .map((line) => {
-      const text = line
-        .replace(cwd + "/", "./")
-        .replace("webpack:/", "")
-        .replace("file://", "")
-        .trim();
-      return {
-        text,
-        internal:
-          (line.includes("node_modules") && !line.includes(".cache")) ||
-          line.includes("internal") ||
-          line.includes("new Promise"),
-      };
-    });
+
+  // Hide details of unhandled/fatal errors in production
+  const hideDetails = !isDev && (error.unhandled || error.fatal);
+
+  const stack = hideDetails
```

Comment:
(we were hiding stack previously. this is a an additional guard added as part of refactor)